Users/randyh/hiptests doc (#3676)
ROCm/rocm-systems: projects/hip/docs/install/build.rst

Reviewer: ```suggestion
After installing and building HIP, you can compile and run your application.
```

Reviewer: ```suggestion
HIP code can be developed on the AMD ROCm platform using HIP-Clang compiler.
```

Reviewer: Link HIP-clang?

Reviewer: ```suggestion
Before building and running HIP, ensure drivers and prebuilt packages are
```

Reviewer: ```suggestion
properly installed on the platform.
```

Reviewer: ```suggestion
As of ROCM 7.1, HIP is integrated into the core ROCm projects that resides in the ``rocm-systems`` monorepository.
In addition, the following components are also part of the monorepository:
```

Reviewer: ```suggestion
   * HIP is installed into ``<ROCM_PATH>``. This can be overridden by setting the ``INSTALL_PREFIX`` as a command option.
```

      <hipamd>/src <hipamd>/include/hip/amd_detail/hip_prof_str.h \
      <hipamd>/include/hip/amd_detail/hip_prof_str.h.new

.. _build-tests:

Build HIP tests
=================================================

> To start you must setup the environment needed to build the HIP tests by setting the ``ROCM_PATH`` environent variable to point to the current installation of ROCm:

Reviewer: ```suggestion
To start, you must set up the environment needed to build the HIP tests by setting the ``ROCM_PATH`` environment variable to point to the current ROCm installation:
```

#. Set the ``HIPTESTS_DIR`` environment variable by running the following from outside the ``hip-tests`` folder: 

   .. code-block:: shell

      export HIPTESTS_DIR="$(readlink -f hip-tests)"
      echo $HIPTESTS_DIR

#. Build HIP all tests from source.

   .. code-block:: shell

      cd "$HIPTESTS_DIR"
      mkdir -p build; cd build
      cmake ../catch -DHIP_PLATFORM=amd -DHIP_PATH=$CLR_DIR/build/install  # or any path where HIP is installed; for example: ``/opt/rocm``
      export ROCM_PATH=/opt/rocm
      make -j$(nproc) build_tests
      ctest # run all tests

   HIP catch source files are found in ``$HIPTESTS_DIR/catch``. Catch tests are built under the folder ``$HIPTESTS_DIR/build/catch_tests``.

   .. note::
      
      To build catch tests with `Address Sanitizer <https://rocm.docs.amd.com/projects/llvm-project/en/latest/conceptual/using-gpu-sanitizer.html>`__ options, add the cmake option ``-DENABLE_ADDRESS_SANITIZER=ON``.

   To run any single built catch test, use this example:

   .. code-block:: shell

      cd $HIPTESTS_DIR/build/catch_tests/unit/texture
      ./TextureTest

#. Build a standalone HIP Catch2 test.

>    HIP Catch2 supports compiling standalone tests using ``amdclang++`` for example:

Reviewer: ```suggestion
   HIP Catch2 supports compiling standalone tests using ``amdclang++``. For example:
```
